app/javascript/packages/document-capture/components/form-steps.jsx

```diff
@@ -79,24 +78,44 @@ export function getStepIndexByName(steps, name) {
   return steps.findIndex((step) => step.name === name);
 }
 
+/**
+ * Returns the first element matched to a field from a set of errors, if exists.
+ *
+ * @param {FormStepError<Record<string,Error>>[]} errors Active form step errors.
+ * @param {Record<string,FieldsRefEntry>} fields Current fields.
+ *
+ * @return {HTMLElement|null|undefined}
```

Comment:
why do we want to return both `null` and `undefined`?

Reply:
> why do we want to return both `null` and `undefined`?

It's not so much a want as much as TypeScript being strict on our behalf:

```
app/javascript/packages/document-capture/components/form-steps.jsx:93:5 - error TS2322: Type 'HTMLElement | null | undefined' is not assignable to type 'HTMLElement | undefined'.
  Type 'null' is not assignable to type 'HTMLElement | undefined'.

93     return fields[error.field].element;
```

Which is because `FieldsRefEntry.element` is defined as possibly `undefined` or `null`:

https://github.com/18F/identity-idp/blob/4e2ff663418fe01ae2d08c17e9f64c8c3e2922bf/app/javascript/packages/document-capture/components/form-steps.jsx#L55

Which is because it could either be `undefined` by absence of the property being assigned here:

https://github.com/18F/identity-idp/blob/4e2ff663418fe01ae2d08c17e9f64c8c3e2922bf/app/javascript/packages/document-capture/components/form-steps.jsx#L225-L234

Or `HTMLElement` or `null` by the ref callback behavior:

https://github.com/18F/identity-idp/blob/4e2ff663418fe01ae2d08c17e9f64c8c3e2922bf/app/javascript/packages/document-capture/components/form-steps.jsx#L227

Related: https://reactjs.org/docs/refs-and-the-dom.html#adding-a-ref-to-a-dom-element

>React will assign the `current` property with the DOM element when the component mounts, and assign it back to `null` when it unmounts. 

One way we could reduce the amount of type juggling here is to just set `element: null` in the initial assignment of `fields.current[field]`, allowing us to at least limit it to either `HTMLElement` or `null`.

Reply:
> One way we could reduce the amount of type juggling here is to just set `element: null` in the initial assignment of `fields.current[field]`, allowing us to at least limit it to either `HTMLElement` or `null`.

It wasn't quite so simple, since the function's `return` statement is (intentionally) wrapped in a condition, so we're still expecting a possibly-`undefined` return value from `getFieldActiveErrorFieldElement`. But we can coerce a `null` value to `undefined` and change the return type to `HTMLElement|undefined`, which I think is a little more sensible for this type of function anyways, since a consumer of `getFieldActiveErrorFieldElement` shouldn't need to be concerned with the implementation detail of ref elements being assigned as `null`.

Updated in 4ff649abf.